addons/website_links/static/src/js/website_links.js

```diff
@@ -533,6 +535,27 @@ publicWidget.registry.websiteLinks = publicWidget.Widget.extend({
     },
 });
 
+popupWidget.include({
+    /**
+     * @override
+     */
+    _onHideModal: function (){
+        this._super.apply(this, arguments);
+        const nbDays = this.$el.find('.modal').data('consentsDuration');
+        const params = new URLSearchParams(window.location.search);
+        const trackingFields = {
+            'utm_campaign': 'odoo_utm_campaign',
+            'utm_source': 'odoo_utm_source',
+            'utm_medium': 'odoo_utm_medium',
+        };
+        for (const [key, value] of params) {
+            if (key in trackingFields) {
+                setCookie(trackingFields[key], value, nbDays * 24 * 60 * 60, 'required');
```

Comment:
checking on python side:

https://github.com/odoo/odoo/blob/8c298467a478af6111b97a345cd1b4f9ade47a35/addons/utm/models/ir_http.py#L21

we always use 31 days, and we only set the cookie if it is optional cookie is enabled (so I guess it would be more consistent to do the same)